docs/groovy/how-to-guides/keyed-transpose.md

```diff
@@ -0,0 +1,241 @@
+)
+```
+
+Each unique combination of `Date` and `Server` creates a separate row in the output.
+
+## Using multiple aggregations
+
+You can apply multiple aggregations simultaneously. When you do this, column names are prefixed with the aggregation name:
+
+```groovy order=result,source
+import io.deephaven.engine.table.impl.util.KeyedTranspose
+
+source = newTable(
+    stringCol("Product", "Widget", "Widget", "Gadget", "Gadget"),
+    stringCol("Region", "North", "South", "North", "South"),
+    intCol("Sales", 100, 150, 200, 175),
+    doubleCol("Revenue", 1000.0, 1500.0, 2000.0, 1750.0)
+)
+
+result = KeyedTranspose.keyedTranspose(
+    source,
+    List.of(
+        AggSum("TotalSales=Sales"),
+        AggAvg("AvgRevenue=Revenue")
+    ),
+    ColumnName.from("Product"),
+    ColumnName.from("Region")
+)
+```
+
+The resulting columns will be named like `TotalSales_North`, `TotalSales_South`, `AvgRevenue_North`, `AvgRevenue_South`.
+
+## Handling initial groups for ticking tables
+
+When working with ticking (live updating) tables, you may want to ensure all expected columns exist from the start, even if no data has arrived yet. Use the `initialGroups` parameter:
+
+```groovy order=result,source
+import io.deephaven.engine.table.impl.util.KeyedTranspose
+
+source = newTable(
+    stringCol("Date", "2025-08-05", "2025-08-05"),
+    stringCol("Level", "INFO", "INFO"),
+    intCol("NodeId", 10, 10)
+)
+
+// Define all expected combinations of Level and NodeId
+initGroups = newTable(
+    stringCol("Level", "ERROR", "WARN", "INFO", "ERROR", "WARN", "INFO"),
+    intCol("NodeId", 10, 10, 10, 20, 20, 20)
+).join(source.selectDistinct("Date"))
+
+result = KeyedTranspose.keyedTranspose(
+    source, 
+    List.of(AggCount("Count")), 
+    ColumnName.from("Date"),
+    ColumnName.from("Level", "NodeId"), 
+    initGroups
+)
+```
+
+Even though the source only has INFO logs from NodeId 10, the result will include columns for all Level-NodeId combinations specified in `initGroups`.
+
+## Understanding column naming
+
+The `keyedTranspose` operation follows specific rules for naming output columns:
+
+| Scenario                             | Column Naming Pattern         | Example                  |
+| ------------------------------------ | ----------------------------- | ------------------------ |
+| Single aggregation, single column-by | Value from column-by column   | `INFO`, `WARN`           |
+| Multiple aggregations                | Aggregation name + value      | `Count_INFO`, `Sum_WARN` |
+| Multiple column-by columns           | Values joined with underscore | `INFO_10`, `WARN_20`     |
+| Invalid characters                   | Characters removed            | `1-2.3/4` → `1234`       |
+| Starts with number                   | Prefixed with `column_`       | `123` → `column_123`     |
+| Duplicate names                      | Suffix added                  | `INFO`, `INFO2`          |
+
+### Best practice: Sanitize data before transposing
```

Comment:
I'm not sure I'd say this is a best practice. In cases of larger tables, mass string replace operations could be expensive.

A better way to go about this would be to show an example of everything in the table above. Each example can go into its own subsection. Either that, or you can present every item in the table above as a subsection with an example.